src/shogun/lib/SGMatrix.h

```diff
@@ -133,6 +134,15 @@ template<class T> class SGMatrix : public SGReferencedData
 		virtual ~SGMatrix();
 
 #ifndef SWIG // SWIG should skip this parts
+		/** @return a (copied) typed matrix with same content */
+		template <class X>
+		SGMatrix<X> as() const
+		{
+			SGMatrix<X> mat(num_rows, num_cols, true);
```

Comment:
the true here technically is not needed....do you prefer it?

Reply:
nope, I will change it